koku/api/settings/settings.py

```diff
@@ -137,47 +144,68 @@ def _build_tag_key(self):
-                "rightTitle": obtainTagKeysProvidersParams[providerName]["rightLabel"],
-                "noValueTitle": "No enabled tag keys",
-                "noOptionsTitle": "No available tag keys",
-                "filterOptionsTitle": "Filter by available tag keys",
-                "filterValueTitle": "Filter by enabled tag keys",
-                "filterValueText": "Remove your filter to see all enabled tag keys",
-                "filterOptionsText": "Remove your filter to see all available tag keys",
-                "initialValue": enabled,
-                "clearedValue": [],
+            avail_objs.append(
+                {
+                    "label": obtainTagKeysProvidersParams[providerName]["title"],
+                    "hasBadge": "true",
+                    "children": [
+                        {"value": "".join([providerName, "-", tag_key]), "label": tag_key} for tag_key in available
+                    ],
+                }
+            )
+            if enabled:
+                enabled_objs.extend("".join([providerName, "-", tag_key]) for tag_key in enabled)
+
+        dual_list_options = {
+            "isTree": "true",
+            "options": avail_objs,
+            "leftTitle": "Disabled tags/labels",
+            "rightTitle": "Enabled tags/labels",
+            "initialValue": enabled_objs,
+            "clearedValue": [],
+        }
+
+        dual_list_name = f'{"api.settings.tag-management.enabled"}'
+        tags_and_labels = create_dual_list_select(dual_list_name, **dual_list_options)
+
+        # currency settings TODO: only show in dev mode right now
+        if settings.DEVELOPMENT:
+            currency_select_name = f'{"api.settings.currency"}'
+            currency_text_context = "Select the preferred currency to view Cost Information in."
+            currency_title = create_plain_text(currency_select_name, "Currency", "h2")
+            currency_select_text = create_plain_text(currency_select_name, currency_text_context, "h4")
+            currency_options = {
+                "label": "Currency",
+                "options": get_currency_options(),
+                "initialValue": get_selected_currency_or_setup(self.schema),
             }
-            dual_list_name = f"{self._get_tag_management_prefix(providerName)}.enabled"
-            dual_list_title = obtainTagKeysProvidersParams[providerName]["title"]
-            components.append(create_plain_text(f"{SETTINGS_PREFIX}.{providerName}.title", dual_list_title, "h2"))
-            components.append(create_dual_list_select(dual_list_name, **dual_list_options))
-        sub_form_name = f"{SETTINGS_PREFIX}.tag_managment.subform"
-        sub_form_title = "Enable tags and labels"
-        sub_form_fields = [tag_key_text]
-        sub_form_fields.extend(components)
-        sub_form = create_subform(sub_form_name, sub_form_title, sub_form_fields)
+            currency = create_select(currency_select_name, **currency_options)
```

Comment:
It looks odd to have the currency select span the width of the page. 

Check with M. Coker, but I believe the Select is intended to fill the width of its container, but not the entire width of the card?

FYI, the Select components used in our details pages don't expand the width of the page. That's because the width is restricted by the toolbar's "item" container.

In that case, PatternFly's `--pf-c-toolbar__item--Width--base` variable is setting a `min-width` of 232px.

Reply:
This is going to be handled in another PR, I am waiting on DDF to include the dropdown component. The issue here is that the component being used is the PF Select component. There are tow things i am waiting on from DDF project owners

1. There is no obvious way to over ride stying, and if there is can they give me an example.
2. Can the PF dropdown component be added, and is there docs on porting. (If they can not get it in soon rather then later.

I agree it looks odd but at this point this is what we have until either of the two above are taking care of. And it would be nice to get this in.

Reply:
I have gotten the style working here...